src/coord/rmr/conn.c

```diff
@@ -425,11 +426,31 @@ static int MRConn_SendAuth(MRConn *conn) {
   CONN_LOG(conn, "Authenticating...");
 
   // if we failed to send the auth command, start a reconnect loop
-  if (redisAsyncCommand(conn->conn, MRConn_AuthCallback, conn, "AUTH %s %s", clusterConfig.aclUsername, conn->ep.password) ==
-      REDIS_ERR) {
-    MRConn_SwitchState(conn, MRConn_ReAuth);
-    return REDIS_ERR;
+  size_t len = 0;
+
+  if (!IsEnterprise()) {
+    // Take the GIL before calling the internal function getter
+    RedisModule_ThreadSafeContextLock(RSDummyContext);
+    const char *internal_secret = RedisModule_GetInternalSecret(RSDummyContext, &len);
+    RedisModule_ThreadSafeContextUnlock(RSDummyContext);
+
+    sds secret = sdsnewlen(internal_secret, len);
```

Comment:
I do not think you need to create an sds here, see how I did it here: https://github.com/RedisGears/LibMR/pull/56/files#diff-55c2de0fa49d05f6ed8f0c13cacedc85fba5d5739c8360567743f9f740df3179R596